### ISSUE TITLE
Why not integrate this plugin's function into Vue core?

### ISSUE THREAD

**Issue by wonderbeyond** [vuejs/vue-async-data#17]
Maybe loading data asynchronously is more a common case. I think Vue's core should implement this function, instead of making a plugin.

e.g. We can pass `data` a  function which returns a Promise.

**Comment by ianwalter** [comment]
I agree with @wonderbeyond. I find this note in the README misleading:
```
NOTE: you don't need this if you are using vue-router. Use vue-router's route.data hook instead.
```
What if you want to load async data on a Header component or some other component that's not tied to a route?
